_test/test/common/message_export.dart

```diff
@@ -4,4 +4,5 @@
 
 export 'message.dart'
     if (dart.library.io) 'message_io.dart'
-    if (dart.library.html) 'message_html.dart';
+    if (dart.library.html) 'message_html.dart'
+    if (dart.library.ffi) 'message_ffi_without_io.dart';
```

Comment:
This could just be the default import probably too, but either way works. @kevmoo do we have a recommended way to target wasm with conditional imports?

Reply:
NO! @sigmundch is working on that RIGHT NOW.

See https://github.com/dart-lang/sdk/commit/e8a049fd8e3736bad7d4a457fded5e5a224f8c30

After this lands we can check for js_interop & !js

Reply:
(er, after we have a dev SDK that supports this release)

Reply:
I've updated the import chain to check for `dart:js` not being available and `dart:js_interop` being available.

However, citing the 3.5 release announcement:

> We plan on deprecating the old interop APIs (dart:html, dart:js, package:js, etc.) in our next Dart release, and fully discontinuing them later next year. 

This makes it sound like there's soon going to be a `dart2js` version that doesn't support `dart:js` either, which will once again mean that there's no way to distinguish the compilers through conditional imports.